sql/sql_zip_dict.cc

```diff
@@ -94,8 +98,19 @@ int mysql_create_zip_dict(THD* thd, const char* name, ulong name_len,
         error= ER_READ_ONLY_MODE;
         my_error(error, MYF(0));
         break;
+      case HA_CREATE_ZIP_DICT_OUT_OF_MEMORY:
+        error= ER_OUT_OF_RESOURCES;
+        my_error(error, MYF(0));
+        break;
+      case HA_CREATE_ZIP_DICT_OUT_OF_FILE_SPACE:
+        error= ER_RECORD_FILE_FULL;
+        my_error(error, MYF(0), "SYS_ZIP_DICT");
+        break;
+      case HA_CREATE_ZIP_DICT_TOO_MANY_CONCURRENT_TRXS:
+        error= ER_TOO_MANY_CONCURRENT_TRXS;
+        my_error(error, MYF(0));
+        break;
       default:
-        DBUG_ASSERT(0);
```

Comment:
Perhaps debug build should continue to assert so that further nonspecific errors are caught and handled?